File: c/LeetCode_278_FirstBadVersion.c
#include <stdio.h>
#include <stdlib.h>


int firstBadVersion(int n) {
    int low = 0;
    int high = n; 
    int middle = 0;

    while (low < high) {
        middle = low + (high - low) / 2;
        if (isBadVersion(middle)) {
            high = middle;
        } else {
            low = middle + 1;
        }
    }

    return low;
}


int main() {
    printf("%d ", firstBadVersion(5));

    return 0;
}

File: python/LeetCode_278_FirstBadVersion.py
class Solution:
    def firstBadVersion(self, n: int) -> int:
        low = 0
        high = n
        mid = 0

        while (low < high):
            mid = low + (high - low) // 2
            if (isBadVersion(mid)):
                high = mid
            else: 
                low = mid + 1
        return low